subsys/net/lib/nrf_cloud/src/nrf_cloud_agps.c

```diff
@@ -535,6 +769,56 @@ static size_t get_next_agps_element(struct nrf_cloud_apgs_element *element,
 	return len;
 }
 
+int nrf_cloud_agps_request_cell_location(enum cell_based_location_type type)
+{
+	int err;
+	cJSON *data_obj;
+	cJSON *agps_req_obj;
+
+	/* TODO: currently single cell only */
+	ARG_UNUSED(type);
+	agps_req_obj = json_create_req_obj(AGPS_JSON_APPID_VAL_SINGLE_CELL,
+					   AGPS_JSON_MSG_TYPE_VAL_DATA);
+	data_obj = cJSON_AddObjectToObject(agps_req_obj, AGPS_JSON_DATA_KEY);
+
+	if (!agps_req_obj || !data_obj) {
+		err = -ENOMEM;
+		goto cleanup;
+	}
+
+	/* Add modem info to the data object */
+	err = json_add_modem_info(data_obj);
+	if (err) {
+		LOG_ERR("Failed to add modem info to cell loc req: %d", err);
+		goto cleanup;
+	}
+
+	err = json_send_to_cloud(agps_req_obj);
+
+cleanup:
+	cJSON_Delete(agps_req_obj);
+
+	return err;
+}
+
+int nrf_cloud_agps_get_last_cell_location(double *const lat,
+					  double *const lon)
```

Comment:
Why not add the other two available fields, alt and unc?

Reply:
mostly just waiting to see what the review comments say :)
also, not sure if altitude is even valid for cell-based location, waiting to hear from cole.
if all four fields are valid it might be best just to add a struct for the data?

Reply:
Good point on whether it is valid.  If so, yeah, a struct might be better.

Reply:
altitude is not valid, i've removed it.